launchdarkly/resource_environment.go

```diff
@@ -0,0 +1,146 @@
+package launchdarkly
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"sync"
+)
+
+// Since we cannot delete the last environment in a project, we use a hack with a temporary dummy
+// environment that is created when we want to delete the last one. To avoid race conditions when
+// deleting multiple resources (terraform by default runs such operations concurrently), we need
+// to use a mutex.
+var environmentMutex = &sync.Mutex{}
+
+func resourceEnvironment() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceEnvironmentCreate,
+		Read:   resourceEnvironmentRead,
+		Update: resourceEnvironmentUpdate,
+		Delete: resourceEnvironmentDelete,
+
+		Schema: map[string]*schema.Schema{
+			"project_key": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"key": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"color": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceEnvironmentCreate(d *schema.ResourceData, m interface{}) error {
+	client := m.(Client)
+
+	environmentMutex.Lock()
+	defer environmentMutex.Unlock()
+
+	project := d.Get("project_key").(string)
+	name := d.Get("name").(string)
+	key := d.Get("key").(string)
+	color := d.Get("color").(string)
+
+	payload := map[string]string{
+		"name":  name,
+		"key":   key,
+		"color": color,
+	}
+
+	err := client.Post("/projects/"+project+"/environments", payload, map[int]bool{201: true}, nil)
```

Comment:
You could create a helper function for this since this url comes up often. Also, you can use something like `fmt.Sprintf("/projects/%s/, projectVar)`

Reply:
Good point, done.